# Task M9 T01
## Sentiment Analysis and Texts

#### - Exercise 1

Take any English text you want, and calculate the frequency of its words.

In [40]:
import pandas as pd
import nltk
from nltk.probability import FreqDist
from nltk.corpus import stopwords
from nltk.stem import PorterStemmer
from nltk.sentiment import SentimentIntensityAnalyzer

nltk.download('punkt')
nltk.download('stopwords')
nltk.download('vader_lexicon')

with open('Research Bees.txt', 'r') as file:
    text = file.read()

text = text.lower()

tokens = nltk.word_tokenize(text)

fdist_all = FreqDist(tokens)

df_freq_all = pd.DataFrame.from_dict(fdist_all, orient='index', columns=['Frequency'])
df_freq_all.index.name = 'Words'
df_freq_all.reset_index(inplace=True)

print("Frecuecy words")
df_freq_all

Frecuecy words


[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\user\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\user\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!
[nltk_data] Downloading package vader_lexicon to
[nltk_data]     C:\Users\user\AppData\Roaming\nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


Words  Frequency
0           now          1
1    scientists          1
2            in          4
3           the          6
4            uk          1
..          ...        ...
135       males          1
136      longer          1
137     females          1
138     similar          1
139         age          1

[140 rows x 2 columns]

#### - Exercise 2
Remove stopwords and perform stemming on your dataset.

In [44]:
stop_words = set(stopwords.words('english'))
tokens_cleaned = [word for word in tokens if word not in stop_words]

stemmer = PorterStemmer()
tokens_stemmed = [stemmer.stem(word) for word in tokens_cleaned]

fdist_cleaned = FreqDist(tokens_stemmed)

df_freq_cleaned = pd.DataFrame.from_dict(fdist_cleaned, orient='index', columns=['Frecuency'])
df_freq_cleaned.index.name = 'Words'
df_freq_cleaned.reset_index(inplace=True)

print("\nFrecuency words after remove stopwords and stemming:")
df_freq_cleaned


Frecuency words after remove stopwords and stemming:


Words  Frecuency
0   scientist          1
1          uk          1
2      discov          1
3    bumblebe          1
4       enjoy          1
..        ...        ...
89       male          1
90     longer          1
91      femal          1
92    similar          1
93        age          1

[94 rows x 2 columns]

#### - Exercise 3
Perform sentiment analysis on your dataset. 

In [43]:
from nltk.sentiment import SentimentIntensityAnalyzer

text_cleaned = ' '.join(df_freq_cleaned['Words'].tolist())

analyzer = SentimentIntensityAnalyzer()

sentiment_scores = analyzer.polarity_scores(text_cleaned)

print("Sentiment Analysis:")
print(sentiment_scores)

Sentiment Analysis:
{'neg': 0.021, 'neu': 0.836, 'pos': 0.144, 'compound': 0.9098}


Based on these scores, the sentiment is primarily neutral (0.836), with a slightly positive sentiment (0.144) and a very low negative sentiment (0.021). The compound sentiment score (0.9098) indicates an overall positive sentiment.